src/models/gamma_poisson_model.py

```diff
@@ -0,0 +1,402 @@
+
+       g(k, n | alpha, beta, I, I_max) =
+            C(n, k) (I / I_max)^k (1 - I/I_max)^{n-k} f(n | alpha, beta),
+
+    where C(n, k) is the binomial coefficient n!/[k!(n-k)!].
+
+    Therefore, the probability that a randomly chosen user will be reached
+    exactly k times is
+
+       g(k | alpha, beta, I, I_max) =
+            sum_{n=k}^{infinity} g(k, n | alpha, beta, I, I_max).
+
+    Thus, if a total of N viewers can potentially be reached, then the expected
+    number of people who will be reached k times is
+
+       hbar[i] = N * g(k | alpha, beta, I, I_max).
+
+    Note that the value I is determined from the input and is not a
+    model parameter that needs to be fit.  Thus, the apparent number
+    of parameters that need to be fit is four.  However, we reduce
+    the number of parameters to two as follows.  First, we note that
+    the expected value of a Gamma-Poisson distribution with parameters
+    alpha and beta is alpha/beta.  Thus, the total number of
+    impressions I_max is related to the total number of potential
+    viewers N via the relation
+
+      E[I_max] = N * alpha/beta.
+
+    We therefore set N = I_max * beta / alpha.  Next, we note that
+    the size of the total impression inventory I_max is essentially
+    a free parameter.  In fact, along the line I_max * beta = constant,
+    the objective function is nearly constant.  To see this, observe
+    that if the available inventory for each user is doubled, then
+    the probability that any given user will receive k impressions
+    remains essentially constant.  Thus, we set I / I_max to a
+    constant value.  This leaves two remaining parameters to optimize,
+    alpha and beta.
+
+    To estimate these parameters we use a histogram matching approach.
+    That is to say, let h[i] be the number of people that were reached
+    i times, and let hbar[i] be the estimated number of people that
+    would be reached i times, given alpha, beta and I_max, N.  The
+    objective function that we compute is as follows:
+
+      chi2(h, hbar) = \sum_i (h[i] - hbar[i])^2 / hbar[i]
+
+    The perceptive reader might recognize this as the chi-squared
+    statistic.  There are a couple advantages to using the chi-squared
+    objective function.  First, in practice, the parameter estimates
+    that are found often give a nearly exact match to the histograms
+    that are observed.  Second, a low value of the objective function
+    gives some confidence that a statistically good fit has been
+    found.
+    """
+
+    def __init__(self, data: [ReachPoint], max_reach=None):
+        """Constructs a Gamma-Poisson model of underreported count data.
+
+        Args:
+          data:  A list of consisting of a single ReachPoint to which the model is
+            to be fit.
+          max_reach:  Optional.  If specified, the maximum possible reach that can
+            be achieved.
+          regularization_parameter:  Optional.  A float specifying a regularization
```

Comment:
As discussed, ```regularization_parameter``` is no longer need, and the corresponding comment should be deleted.